schema/deploy/shipping/views.sql

```diff
@@ -2195,6 +2196,96 @@ grant select
     to "scan-dashboard-exporter";
 
 
+create or replace view shipping.scan_demographics_v2 as
+
+    with hcov19_presence_absence as (
+        -- Collapse potentially multiple hCoV-19 results
+        select distinct on (sample_id)
+            sample.identifier as sample,
+            presence_absence_id,
+            case
+              when present is true then 'positive'
+              when present is false then 'negative'
+              when present is null then 'positive'
+            end as hcov19_result,
+            pa.created::date as hcov19_result_release_date
+        from
+            warehouse.presence_absence as pa
+            join warehouse.target using (target_id)
+            join warehouse.organism using (organism_id)
+            join warehouse.sample using (sample_id)
+        where
+            organism.lineage <@ 'Human_coronavirus.2019'
+            and pa.details @> '{"assay_type": "Clia"}'
+            and not control
+            -- We shouldn't be receiving these results from Samplify, but they
+            -- sometimes sneak in. Be sure to block them from this view so as
+            -- to not return inaccurate results to participants.
+            and target.identifier not in ('COVID-19_Orf1b', 'COVID-19-S_gene')
+        /*
+          Keep only the most recent push. According to Lea, samples are only
+          retested if there is a failed result. A positive, negative, or
+          indeterminate result would not be retested.
+
+          https://seattle-flu-study.slack.com/archives/CV1E2BC8N/p1584570226450500?thread_ts=1584569401.449800&cid=CV1E2BC8N
+        */
+        order by
+            sample_id, presence_absence_id desc
+    ),
+
+    king_county_demographics as (
+        select
+            encountered_week,
+            sex,
+            age_range_decade,
+            age_range_decade_lower,
+            age_range_decade_upper,
+            race,
+            hispanic_or_latino,
+            income,
+            sample
+        from shipping.scan_encounters_v1
+        -- Limit to King County participants only since the dashboard
+        -- is only for King County.
+        where puma in (
+          '5311601', '5311602', '5311603', '5311604',
+          '5311605', '5311606', '5311607', '5311608',
+          '5311609', '5311610', '5311611', '5311612',
+          '5311613', '5311614', '5311615', '5311616')
```

Comment:
where'd you get this list of King County PUMAs by the way?

Reply:
Found here: https://www.census.gov/geographies/reference-maps/2010/geo/2010-pumas/washington.html

I was actually wondering if we should formalize the PUMAs -> county hierarchy in the locations table.

Reply:
That's not a bad idea, at the very least for WA counties or King & Yakima counties.